test/unit/dom/wait_for_test.dart

```diff
@@ -201,6 +201,16 @@ void main() {
                 hasToStringValue(contains('The callback must return a non-null Element.')),
               )));
         }, timeout: asyncQueryTestTimeout);
+
+        test('throws a timeout error based on a provided waitForElementToBeRemoved() timeout duration', () async {
+          expect(
+              () => rtl.waitForElementToBeRemoved(() => elementThatWillBeRemovedAfterDelay, container: view.container, 
+                timeout: Duration(milliseconds: 1)),
```

Comment:
This test is more of a sanity check. Since `waitForElementToBeRemoved` and `waitFor` share the same timeout duration, I'm testing that `waitForElementToBeRemoved` throws an error with the timeout value it's explicitly given (on line 208). This works even _without_ our default timeout in `wait_for.dart`, but we can be confident it's using the timeout in `waitForElementToBeRemoved` when printing the error.

We could also arbitrarily set `waitForElementToBeRemoved` timeout duration to be different from `waitFor`, but that seems unnecessary.

Open to feedback on this one. Gonna think more about it over the weekend. The brain fog is real